freeipa/src/main/java/com/sequenceiq/freeipa/util/KrbKeySetEncoder.java

```diff
@@ -0,0 +1,88 @@
+package com.sequenceiq.freeipa.util;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.ActorKerberosKey;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+
+import org.bouncycastle.asn1.ASN1Encodable;
+import org.bouncycastle.asn1.DERInteger;
+import org.bouncycastle.asn1.DEROctetString;
+import org.bouncycastle.asn1.DERSequence;
+import org.bouncycastle.asn1.DERTaggedObject;
+
+public final class KrbKeySetEncoder {
+
+    private static final int TAG_0 = 0;
+
+    private static final int TAG_1 = 1;
+
+    private static final int TAG_2 = 2;
+
+    private static final int TAG_3 = 3;
+
+    private static final int TAG_4 = 4;
+
+    private static final int NUM_KEYS = 2;
+
+    private KrbKeySetEncoder() {
+
+    }
+
+    private static DERSequence makeSalt(int type, String salt) throws UnsupportedEncodingException {
+        return new DERSequence(new ASN1Encodable[]{
+            new DERTaggedObject(true, TAG_0, new DERInteger(type)),
+            new DERTaggedObject(true, TAG_1, new DEROctetString(salt.getBytes("UTF-8")))
+        });
+    }
+
+    private static DERSequence makeEncryptionKey(int enctype, byte[] value) {
+        return new DERSequence(new ASN1Encodable[]{
+            new DERTaggedObject(true, TAG_0, new DERInteger(enctype)),
+            new DERTaggedObject(true, TAG_1, new DEROctetString(value))
+        });
+    }
+
+    private static DERSequence makeKrbKey(DERSequence salt, DERSequence key) {
+        return new DERSequence(new ASN1Encodable[] {
+            new DERTaggedObject(true, TAG_0, salt),
+            new DERTaggedObject(true, TAG_1, key)
+        });
+    }
+
+    public static String getASNEncodedKrbPrincipalKey(List<ActorKerberosKey> keys) throws IOException {
+        if (keys == null || keys.size() < NUM_KEYS) {
+            throw new IllegalArgumentException("Invalid kerberos keys provided");
+        }
+
+        ASN1Encodable[] asn1Encodables = new ASN1Encodable[NUM_KEYS];
```

Comment:
same question as before: why not use keys.size()? there's no reason this code has to care how many keys are encoded (it would be fine for us to have only one key, if we decide that AES256 is bad in the future or whatever)